ldap-authenticator/src/main/java/org/xwiki/contrib/ldap/XWikiLDAPConfig.java

```diff
@@ -325,6 +331,11 @@ public String getLDAPParam(String name, String cfgName, String def)
             param = this.cfgConfigurationSource.getProperty(cfgName);
         }
 
+        // Look in the default configuration that might be provided by the enabled authenticator
+        if (param == null || "".equals(param)) {
```

Comment:
1) as you can see in the code handling default value "" is a real value so it does not make sense to test it here
2) this is not the right place, the passed "def" should have priority over the standard default so you should look at the map only if def is null in the following if.